In [172]:
#Load all necesaary modules
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
#from sklearn.cross_validation import train_test_split
from sklearn.model_selection import train_test_split
from sklearn import metrics

# visualization
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [173]:
#Load data with pandas 
# read data into a DataFrame
data = pd.read_csv("drinks.csv")

#DAta Exploration
#get the first five rows
data.head()


country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   
3      Andorra            245              138            312   
4       Angola            217               57             45   

   total_litres_of_pure_alcohol continent  
0                           0.0        AS  
1                           4.9        EU  
2                           0.7        AF  
3                          12.4        EU  
4                           5.9        AF

In [174]:
data.shape

(193, 6)

In [175]:
data.describe()

beer_servings  spirit_servings  wine_servings  \
count     193.000000       193.000000     193.000000   
mean      106.160622        80.994819      49.450777   
std       101.143103        88.284312      79.697598   
min         0.000000         0.000000       0.000000   
25%        20.000000         4.000000       1.000000   
50%        76.000000        56.000000       8.000000   
75%       188.000000       128.000000      59.000000   
max       376.000000       438.000000     370.000000   

       total_litres_of_pure_alcohol  
count                    193.000000  
mean                       4.717098  
std                        3.773298  
min                        0.000000  
25%                        1.300000  
50%                        4.200000  
75%                        7.200000  
max                       14.400000

In [182]:
data['continent'].value_counts()

AF    53
EU    45
AS    44
OC    16
SA    12
Name: continent, dtype: int64

In [183]:
data['country'].value_counts()

St. Vincent & the Grenadines    1
Colombia                        1
Lithuania                       1
Macedonia                       1
Ireland                         1
                               ..
Sri Lanka                       1
San Marino                      1
Libya                           1
Somalia                         1
Laos                            1
Name: country, Length: 193, dtype: int64

QUESTION TO TACKLE:

Given the various alcoholic servings by country or continent, what is the effect of the servings unto the total litres of pure alcohol that will be consumed?

In [176]:
#Prediction target => output variable
y = data['total_litres_of_pure_alcohol']

#Feature set => Input variables
"""
For the sake of linear regression we'll only play around with quantitative data and discard the qualitative data
"""
#The commented implementation appears to be too rigid. Hence, included the dynamic dropping of qualitative columns
X = data.drop(['country', 'continent'], axis=1)

#X = data._get_numeric_data().drop(['total_litres_of_pure_alcohol'], axis=1)

In [177]:
#Define the model
model = LinearRegression()

#Data splitting => test and training data. It may also be the validation and training data. It will randomly and auto pick the training size
X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.7, random_state=20 )

In [178]:
#Fit the model
model.fit(X_train, y_train)

LinearRegression()

In [179]:
#Make prediction based on the trained model
y_pred=model.predict(X_test)

In [180]:
#Evaluation metrics as accuracy is often not vital in the regression problem => loss functions
metrics.mean_absolute_error(y_test, y_pred)

2.590839420396809e-15

In [181]:
#Alternative that engulfs model selection, fitting, prediction and evaluation metrics process
def weigh_effect(X, y):
    
    #Define the model
    model = LinearRegression()
    
    #Data splitting => test and training data. It may also be the validation and training data. It will randomly and auto pick the training size
    X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.7, random_state=20 )
    
    #Fit the model
    model.fit(X_train, y_train)
    
    #Make prediction based on the trained model
    y_pred=model.predict(X_test)
    
    #Evaluate metrics as accuracy if often not vital in the regression problem
    return metrics.mean_absolute_error(y_test, y_pred)
    

METRIC RESULTS FOR EITHER OF THE THREE FEATURES AT RANDOM ARE AS FOLLOWS:
    1. beer_servings and spirit_Servings returns 2.3558
    2. beer_servings and wine_Servings returns 1.5155
    3. wine_servings and spirit_Servings returns 3.1837
    4. spirit_servings returns 1.4515
    5. beer_servings returns 6.115
    6. wine_servings returns 1.066 
    7. beer_servings, wine_servings and spirit_servings returns 2.5908



Recommendation:
    
It is clear that wine_servings has a low error count. seconded by spirit_servings and finally the beer_servings. Hence, more wine servings ld be included in order to produce more litres of pure wine against the other combinations. 